### DISCUSSION TITLE
ChatDA: Agentic NL2SQL, Automated Data Visualization and Analysis with LLM

### DISCUSSION THREAD

**Discussion by ruohuawang** [InternLM/Tutorial#698] [project]
We propose a LLM-based agent system that perform a pipeline of data fetching, visualization and analysis.
The procedure is described as follows: 1. the system first accepts the user's natural language query; 2. NL2Metric: a encoder-only/encoder-decoder LLM (e.g., BERT or T5 ) then perform natural language understanding (NLU) tasks, e.g. named entity recognition and intention idetification, transforming user's query into metrics in json format; 3.retrieval augmented generation (RAG): borrowing ideas from Huixiangdou, we use bce-reranker as retrieval module to do field-level RAG; 4.Metric2SQL: a decoder-only LLM (e.g. InternLM2 14B) then generats SQL; 5. multiple conversations: reflection and revision; 6. excecution of SQL; 7. data visualization and analysis using InternLM2's agent skills. In the last, we get a report with charts and tables based on the given user's query.
We use the following inference/deployment framework: lmdeploy, gradio, and lagent (xtuner if needed).

**Comment by GoXian** [comment]
请问项目地址在哪
